### PR TITLE
Updated Dockerfile to follow the build instructions from rizin README.

### PR DESCRIPTION
**Pull request by 0pendev**
Signed-off-by: Francisco Freitas <contact@ffreitas.io>

 

**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/rizinbook) with the relevant information (if needed)

**Detailed description**

Looking at the Dockerfile I saw that it was not using the rizin build instructions to use meson. Another problem was that the project request a meson version that is not available in debian:10. I got the error message : 
`meson.build:1:0: ERROR:  Meson version is 0.49.2 but project requires >=0.50.1.`

I moved the image to OpenSUSE/leap as I usually build with this image.

I did not added rz-ghidra in the build because I did not know if it was usefull. But it could make sense to add it by default like in Cutter.

I removed the addition to the sudo group. If a user needs to install a new package he can just use the argument --user root when starting the container. If privileged are needed (for debugging for example) the user just needs to use the argument --privileged when starting the container. 

**Test plan**

I think I might need more help here. I simply ran a simple analysis on /bin/ls : 
image

**Closing issues**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -31,13 +31,13 @@
 #
 
 # Using debian 10 as base image.
-FROM debian:10
+FROM opensuse/leap:15.2
```
Comment (wargio) [inline Dockerfile]: not super sure about the distro choice (just an issue regarding the image size)

**Review by wargio — APPROVED** [review]
Looks great! i before merging this i just want to check if for the others is ok this distro, but besides that, thank you :)

**Review by ret2libc — COMMENTED** [review]
Thanks a lot! I was actually going to work on this but postponed until tomorrow... The way i bypassed the meson problem was to just install meson and ninja from pip, as that gets you an updated one. Opensuse could work as well, though probably most people are more familiar with Debian/Ubuntu than with suse

File: Dockerfile (hunk 1 above)
Comment (XVilka) [reply Dockerfile]: I think it's better to stick to Debian here. Or Ubuntu, at least.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (thestr4ng3r) [reply Dockerfile]: Even better alpine

Comment (0pendev) [reply Dockerfile]: Debian is not an option as the meson version is not working. I'll check alpine to try to limit its size.

Comment (0pendev) [reply Dockerfile]: Might try installing from pip as suggested by @ret2libc

Comment (ret2libc) [reply Dockerfile]: Alpine with pip should work, yeah!

File: Dockerfile [hunk 2]
```diff
@@ -60,48 +60,37 @@ VOLUME ["/mnt"]
...
+	apt-get install -y \
+	gcc \
+	git \
+	python3-pip \
+	${with_arm64_as:+binutils-aarch64-linux-gnu} \
+	${with_arm32_as:+binutils-arm-linux-gnueabi} \
+	${with_ppc_as:+binutils-powerpc64le-linux-gnu} && \
+	pip3 install meson ninja && \
+	cd /mnt && \
+	git clone https://github.com/rizinorg/rz-pipe && \
```
Comment (ret2libc) [inline Dockerfile]: set `RZ_PIPE_PY_VERSION` default to `master` and use 
```suggestion
	git clone -b "$RZ_PIPE_PY_VERSION" https://github.com/rizinorg/rz-pipe && \
```

File: Dockerfile [hunk 3]
```diff
@@ -60,48 +60,37 @@ VOLUME ["/mnt"]
...
 # Cleanup
-RUN DEBIAN_FRONTEND=noninteractive dpkg --add-architecture i386 && \
-  apt-get update && \
-  apt-get install -y \
-  curl \
-  wget \
-  gcc \
-  git \
-  bison \
-  pkg-config \
```
Comment (ret2libc) [inline Dockerfile:71]: I think this is needed?

File: Dockerfile (hunk 2 above)
Comment (0pendev) [reply Dockerfile]: Forgot to add the branch. I'm adding it.

File: Dockerfile (hunk 3 above)
Comment (0pendev) [reply Dockerfile:71]: bison, pkg-config are not used when building with meson (or at leasts it builds and the rizin binary seems  ok).

I only install those packages :  gcc  git python3-pip.

Do you have a use case where bison and pkg-config are used to build when using meson ?

**Comment by 0pendev** [comment]
As people where not that happy with the base image change I went back to debian.


Alpine does not package binutils for the architectures specified in the Dockerfile.
```
/ # apk search binutils
binutils-gold-2.34-r1
mingw-w64-binutils-2.34-r0
binutils-avr-doc-2.34-r0
mingw-w64-binutils-doc-2.34-r0
binutils-doc-2.34-r1
binutils-dev-2.34-r1
binutils-2.34-r1
binutils-avr-2.34-r0
```

File: Dockerfile (hunk 3 above)
Comment (ret2libc) [reply Dockerfile:71]: Then it's good :) I thought it meson/ninja needed that, but probably it's just to build against the libraries.

**Comment by wargio** [comment]
For me debian is ok!! Thank you again for this contrib

File: Dockerfile (hunk 2 above)
Comment (ret2libc) [reply Dockerfile]: Just saw the review request. Did you already add this?

Comment (wargio) [reply Dockerfile]: @ret2libc search for `$RZ_VERSION`

Comment (ret2libc) [reply Dockerfile]: No, that is different @wargio . This should use `RZ_PIPE_PY_VERSION`, not `RZ_VERSION`.

**Comment by XVilka** [comment]
@ret2libc I think the docker CI job should be triggered on the change of Dockerfile and scripts that are used by it.

File: Dockerfile (hunk 2 above)
Comment (0pendev) [reply Dockerfile]: Sorry I thought I already added it. It's done now.

**Review by ret2libc — APPROVED** [review]
Thanks a lot!

**Comment by ret2libc** [comment]
Merged!

**Comment by 0pendev** [comment]
> Merged!

Nice !

**Pull request merged**